component/s3/file/s3.go

```diff
@@ -0,0 +1,223 @@
+package s3
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	aws_config "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+	"github.com/grafana/agent/pkg/river"
+	"golang.org/x/net/context"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "s3.file",
+		Args:    Arguments{},
+		Exports: Exports{},
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// S3 handles reading content from a file located in S3
+type S3 struct {
+	mut        sync.Mutex
```

Comment:
Not incredibly happy with the amount of mutex lock/unlocks I have, but not sure if there is a better way